Makefile

```diff
@@ -228,8 +230,15 @@ test-integration: clean-cov generate fmt vet ginkgo ## Run Integration tests.
 		--coverpkg $(INTEGRATION_COVER_PKGS) \
 		--output-dir $(PROJECT_PATH)/coverage/integration \
 		--coverprofile cover.out \
-		--fail-fast \
 		-v \
+		--compilers=$(INTEGRATION_TEST_NUM_CORES) \
+		--procs=$(INTEGRATION_TEST_NUM_PROCESSES) \
+		--randomize-all \
+		--randomize-suites \
+		--fail-on-pending \
+		--keep-going \
```

Comment:
```
--keep-going 
    If set, failures from earlier test suites do not prevent later test suites
    from running.
```

Not an strong opinion here. But if one test fails, I prefer the test suite to end. I do not need to know how many tests are failing. I fix the failing test and trying again the whole suite.

Reply:
My preference is to use the `keep-going` flag instead of the `fail-fast`. Although it may be not an issue to using the `fail-fast` flag now that the integration test completes much faster,  using `keep-going` can tell a developer all tests that were effected by a change rather than fixing them sequentially 🤔 

Can revert back to `--fail-fast` if you feel strongly about this. I don't see it as a major downside to keeping to `fail-fast` if we can now run integration tests fairly quickly

Reply:
Let's give it a try.

If the test duration was too long, I think it is best `--fail-fast`. I would not like to wait 1 hour just to see all tests that are failing.  If it is a matter of few minutes, I am ok with keep-going.

Reply:
Tried to use `-p` in https://github.com/Kuadrant/limitador-operator/pull/134/commits/5a59106568bebef372b8a6d3a7d1be6170c3fe5b and looks like it's using 4 processes for the tests - https://github.com/Kuadrant/limitador-operator/actions/runs/9081747820/job/24956406772?pr=134 which matches the number specified at https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories

![image](https://github.com/Kuadrant/limitador-operator/assets/24636860/7bd7e9f3-549b-412e-baef-28d9bfd7f317)

It's about 1.5 mins slower than previous where we able to specify to run using 10 processes instead 🤔 

Probably will revert to the previous commit since it looks like github CI is able to support it unless you want to go with `-p` instead